docs/api_guide/plot_06_sklearn_pipeline_cv_demo.py

```diff
@@ -0,0 +1,473 @@
+X = np.random.uniform(low=0, high=1, size=(1000, 1))
+# observed counts, shape (n_samples,)
+rate = 2 * (
+    scipy.stats.norm.pdf(X, scale=0.1, loc=0.25)
+    + scipy.stats.norm.pdf(X, scale=0.1, loc=0.75)
+)
+y = np.random.poisson(rate).astype(float).flatten()
+
+# %%
+fig, ax = plt.subplots()
+ax.scatter(X.flatten(), y, alpha=0.2)
+ax.set_xlabel("input")
+ax.set_ylabel("spike count")
+sns.despine(ax=ax)
+
+# %%
+# ### Converting NeMoS `Basis` to a transformer
+# In order to use NeMoS `Basis` in a pipeline, we need to convert it into a scikit-learn transformer. This can be achieved through the `TransformerBasis` wrapper class.
+#
+# Instantiating a `TransformerBasis` can be done using the constructor directly or with `Basis.to_transformer()`:
+
+# %%
+bas = nmo.basis.RaisedCosineBasisLinear(5, mode="conv", window_size=5)
+trans_bas_a = nmo.basis.TransformerBasis(bas)
+trans_bas_b = bas.to_transformer()
+
+# %%
+# `TransformerBasis` provides convenient access to the underlying `Basis` object's attributes:
+
+# %%
+print(bas.n_basis_funcs, trans_bas_a.n_basis_funcs, trans_bas_b.n_basis_funcs)
+
+# %%
+# We can also set attributes of the underlying `Basis`. Note that -- because `TransformerBasis` is created with a copy of the `Basis` object passed to it -- this does not change the original `Basis`, and neither does changing the original `Basis` change `TransformerBasis` we created:
+
+# %%
+trans_bas_a.n_basis_funcs = 10
+bas.n_basis_funcs = 100
+
+print(bas.n_basis_funcs, trans_bas_a.n_basis_funcs, trans_bas_b.n_basis_funcs)
+
+# %%
+# ### Creating and fitting a pipeline
+# We might want to combine first transforming the input data with our basis functions, then fitting a GLM on the transformed data.
+#
+# This is exactly what `Pipeline` is for!
+
+# %%
+pipeline = Pipeline(
+    [
+        (
+            "transformerbasis",
+            nmo.basis.TransformerBasis(nmo.basis.RaisedCosineBasisLinear(6)),
+        ),
+        (
+            "glm",
+            nmo.glm.GLM(regularizer_strength=0.5, regularizer="Ridge"),
+        ),
+    ]
+)
+
+pipeline.fit(X, y)
+
+# %%
+# Note how NeMoS models are already scikit-learn compatible and can be used directly in the pipeline.
+#
+# Visualize the fit:
+
+# %%
+
+# Predict the rate.
+# Note that you need a 2D input even if x is a flat array.
+# We are using expand dim to add the extra-dimension
+x = np.expand_dims(np.sort(X.flatten()), axis=1)
+predicted_rate = pipeline.predict(x)
+
+# %%
+fig, ax = plt.subplots()
+
+ax.scatter(X.flatten(), y, alpha=0.2, label="generated spike counts")
+ax.set_xlabel("input")
+ax.set_ylabel("spike count")
+
+
+ax.plot(
+    x,
+    predicted_rate,
+    label="predicted rate",
+    color="tab:orange",
+)
+
+ax.legend()
+sns.despine(ax=ax)
+
+# %%
+# There is some room for improvement, so in the next section we'll use cross-validation to tune the parameters of our pipeline.
+
+# %%
+# ### Select the number of basis by cross-validation
```

Comment:
I think, for the first example, we should show what the "manual way" of doing this would be (maybe in an admonition). That will make it both more explicit what this is doing, as well as clarifying how much boilerplate this saves you

Reply:
pretty impressive reduction in lines of code